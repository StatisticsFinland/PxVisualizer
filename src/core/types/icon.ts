export type TIcon = typeof allIcons[number];
export const allIcons = ['Bars', 'Times', 'Download', 'ExternalLink', 'Copy'] as const;