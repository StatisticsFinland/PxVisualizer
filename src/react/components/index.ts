export * from "./chart";
export * from "./ErrorBoundary";
export * from "./globalStyle";
export * from "./icon";
export * from "./burgerMenu";
