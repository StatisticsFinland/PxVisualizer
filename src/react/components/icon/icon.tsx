import React from 'react';
import { TIcon } from '../../../core/types/icon';

interface IIconProps {
    icon: TIcon;
    inheritColor?: boolean;
};

/* no need for tests for icon svg markups */
/* istanbul ignore next */
export const Icon: React.FC<IIconProps> = ({icon, inheritColor=false}) => { // NOSONAR
    switch(icon) { //NOSONAR
        case "Bars":
            return (
                <svg aria-hidden={true} xmlns="http://www.w3.org/2000/svg" width="39.91mm" height="29.92mm" viewBox="0 0 113.12 84.82">
                    <path { ... inheritColor ? {} : {style: {fill: '#1A3061'}}} d="m56.51,14.13c-16.33,0-32.67.01-49,0-5.37,0-8.89-4.83-6.99-9.59C1.63,1.76,3.77.26,6.76.01c.22-.02.44,0,.66,0,32.74,0,65.49,0,98.23,0,4.3,0,7.45,2.99,7.47,7.01.02,4.11-3.15,7.1-7.6,7.11-16.33.01-32.67,0-49,0Z"/>
                    <path { ... inheritColor ? {} : {style: {fill: '#1A3061'}}} d="m56.55,49.47c-16.41,0-32.82.01-49.22,0-3.78,0-6.77-2.56-7.26-6.09-.49-3.53,1.78-6.9,5.35-7.81.77-.2,1.6-.21,2.4-.21,32.52,0,65.04-.01,97.57,0,4.51,0,7.69,2.89,7.74,6.97.04,4.19-3.16,7.14-7.79,7.15-16.26,0-32.52,0-48.78,0Z"/>
                    <path { ... inheritColor ? {} : {style: {fill: '#1A3061'}}} d="m56.43,84.82c-16.33,0-32.67,0-49,0-3.68,0-6.45-2.17-7.25-5.56-.73-3.09.92-6.54,3.92-7.84,1.04-.45,2.27-.69,3.41-.69,32.74-.04,65.49-.03,98.23-.02,4.27,0,7.41,3.06,7.39,7.09-.02,4.02-3.17,7.02-7.47,7.02-16.41.02-32.82,0-49.22,0Z"/>
                </svg>
            );
        case "Times":
            return (
                <svg aria-hidden={true} xmlns="http://www.w3.org/2000/svg" width="26.97mm" height="26.97mm" viewBox="0 0 76.44 76.44">
                    <path { ... inheritColor ? {} : {style: {fill: '#1A3061'}}} d="m0,38.09C.02,17.02,17.22-.04,38.39,0c21.08.04,38.09,17.21,38.05,38.42-.04,21-17.21,38.04-38.31,38.01C17.04,76.41-.02,59.26,0,38.09ZM38.25,5.95c-17.86-.02-32.27,14.33-32.31,32.17-.04,17.94,14.33,32.35,32.25,32.37,17.87.02,32.23-14.28,32.3-32.17.07-17.92-14.31-32.35-32.24-32.37Z"/>
                    <path { ... inheritColor ? {} : {style: {fill: '#1A3061'}}} d="m38.4,34c2.9-2.94,5.58-5.67,8.27-8.39.93-.94,2.13-1.38,3.25-.73.8.47,1.7,1.46,1.78,2.29.09.92-.48,2.12-1.14,2.85-2.18,2.38-4.54,4.6-6.83,6.89-.35.35-.68.71-1.07,1.12,2.67,2.68,5.35,5.26,7.88,7.99.67.72,1.28,1.95,1.16,2.84-.12.88-1.05,1.9-1.89,2.37-1.03.57-2.22.16-3.12-.75-2.31-2.33-4.65-4.63-6.96-6.97-.38-.39-.66-.87-1.36-1.8-.71.95-1.11,1.64-1.65,2.19-2.12,2.17-4.27,4.32-6.44,6.44-1.41,1.38-3.16,1.46-4.38.28-1.26-1.22-1.2-3.02.24-4.48,2.27-2.31,4.56-4.58,6.87-6.85.38-.38.88-.64,1.42-1.03-2.89-2.89-5.5-5.47-8.07-8.1-.53-.55-1.05-1.24-1.24-1.96-.35-1.32.14-2.48,1.38-3.16,1.28-.71,2.53-.5,3.56.5,2.32,2.26,4.59,4.55,6.88,6.84.46.46.86.96,1.47,1.64Z"/>
                </svg>
            );
        case "Download":
            return (
                <svg aria-hidden={true} xmlns="http://www.w3.org/2000/svg" width="33.64mm" height="33.64mm" viewBox="0 0 95.35 95.37">
                    <path { ... inheritColor ? {} : {style: {fill: '#1A3061'}}} d="m41.74,51.99c0-1.25,0-1.92,0-2.59,0-14.4,0-28.79,0-43.19,0-3.27,2.24-5.84,5.28-6.17,3.01-.33,5.8,1.65,6.48,4.66.16.71.14,1.48.14,2.22,0,14.15,0,28.29,0,42.44v2.34c.53-.42.91-.67,1.23-.98,3.12-3.11,6.23-6.23,9.35-9.34,2.74-2.72,6.26-2.87,8.75-.4,2.49,2.46,2.39,6.06-.3,8.75-6.79,6.81-13.59,13.61-20.4,20.4-2.92,2.91-6.25,2.92-9.15.03-6.81-6.79-13.61-13.59-20.4-20.4-2.71-2.72-2.84-6.26-.35-8.75,2.48-2.49,6.03-2.36,8.75.34,3.42,3.4,6.82,6.83,10.61,10.64Z"/>
                    <path { ... inheritColor ? {} : {style: {fill: '#1A3061'}}} d="m47.67,95.37c-13.83,0-27.66,0-41.49,0-3.18,0-5.7-2.16-6.12-5.13-.42-2.98,1.5-5.82,4.51-6.59.65-.17,1.35-.18,2.02-.18,27.41,0,54.83,0,82.24,0,3.8,0,6.48,2.43,6.52,5.88.04,3.53-2.66,6.02-6.56,6.03-13.71,0-27.41,0-41.12,0Z"/>
                </svg>
            );
        case "ExternalLink":
            return (
                <svg aria-hidden={true} xmlns="http://www.w3.org/2000/svg" width="88.84mm" height="79.62mm" viewBox="0 0 251.84 225.69">
                    <path { ... inheritColor ? {} : {style: {fill: '#1A3061'}}} d="m125.89,225.68c-22.84,0-45.67.03-68.51,0C23.18,225.63.03,202.4.01,168.14,0,131.14,0,94.13.01,57.13.04,23.53,23.35.14,56.85.1c23.67-.03,21.34.02,45.01-.03,4.75,0,8.96,1.04,11.87,5.09,2.64,3.67,3.08,7.7,1.25,11.83-2.01,4.53-5.59,7-10.57,7.06-9,.11-12,.07-21,.07-15.84,0-11.67-.05-27.5.02-18.82.08-31.81,12.98-31.84,31.74-.06,38-.06,76.01,0,114.01.03,18.74,13.04,31.7,31.86,31.72,46.67.05,93.34.06,140.01,0,18.79-.02,31.71-12.99,31.79-31.8.07-17,0-14,.03-31.01.02-8.3,4.75-13.75,11.83-13.81,7.07-.06,12.12,5.48,12.16,13.61.07,17.5.18,15.01-.02,32.51-.34,30.71-23.87,54.02-54.81,54.56h-71.01l-.02.02Z"/>
                    <path { ... inheritColor ? {} : {style: {fill: '#1A3061'}}} d="m207.19,24.08c-6.95,0-23.9.04-30.85,0-8.12-.06-13.27-4.55-13.44-11.6-.17-7.19,5.18-12.33,13.32-12.38,17.33-.11,44.67-.12,62,0,8.48.06,13.45,4.99,13.5,13.37.12,16.83.12,43.67,0,60.5-.06,7.69-5.4,13.04-12.34,12.87-6.83-.17-11.53-5.35-11.61-12.97-.08-7.63-.02-25.26-.02-34.12-2.09,1.8-3.42,2.88-4.68,4.03-26.76,24.57-53.52,49.15-80.28,73.72-1.6,1.47-3.13,3-4.82,4.36-5.57,4.47-12.54,3.94-16.98-1.2-4.31-4.98-4.06-11.78,1.06-16.6,10.92-10.28,22.03-20.36,33.07-30.51,16.56-15.23,33.13-30.45,49.72-45.65.97-.89,2.15-1.56,3.23-2.33-.29-.5-.58-.99-.87-1.49h0Z"/>
                </svg>
            );
        case "Copy":
            return (
                <svg aria-hidden={true} xmlns="http://www.w3.org/2000/svg" width="31.19mm" height="31.21mm" viewBox="0 0 88.41 88.48">
                    <path { ... inheritColor ? {} : {style: {fill: '#1A3061'}}} d="m20.67,54.61c0-7.55,0-15.1,0-22.65,0-6.79,4.38-11.2,11.13-11.21,15.16-.01,30.33-.01,45.49,0,6.75,0,11.12,4.42,11.12,11.22,0,15.1,0,30.2,0,45.3,0,6.79-4.39,11.21-11.13,11.21-15.16,0-30.33,0-45.49,0-6.66,0-11.1-4.41-11.12-11.03-.02-7.61,0-15.22,0-22.83Zm60.78.1c0-7.67,0-15.34,0-23.01,0-2.72-1.23-3.98-3.89-3.99-15.34,0-30.68,0-46.02,0-2.67,0-3.89,1.26-3.89,3.98,0,15.28,0,30.55,0,45.83,0,2.79,1.23,3.99,4.07,3.99,15.22,0,30.44,0,45.65,0,2.91,0,4.08-1.21,4.08-4.17,0-7.55,0-15.09,0-22.64Z"/>
                    <path { ... inheritColor ? {} : {style: {fill: '#1A3061'}}} d="m13.66,60.85v6.64C6.35,69.22.02,64.29.01,56.9c-.02-15.34-.02-30.69,0-46.03C.02,4.48,4.36.11,10.73.1c15.1-.02,30.21.19,45.3-.1,6.89-.13,12.81,4.89,11.49,13.58h-6.74c0-.95.03-1.91,0-2.86-.08-2.34-1.35-3.69-3.66-3.69-15.53-.02-31.06-.02-46.59,0-2.36,0-3.58,1.27-3.58,3.7-.02,15.4-.02,30.81,0,46.21,0,2.65,1.32,3.87,4.01,3.91.85.01,1.7,0,2.71,0Z"/>
                </svg>
            );
        default:
            return (
                <span>No icon for {icon}
                <svg aria-hidden={true} xmlns="http://www.w3.org/2000/svg" width="26.97mm" height="26.97mm" viewBox="0 0 76.45 76.44">
                    <path { ... inheritColor ? {} : {style: {fill: '#1A3061'}}} d="m0,38.1C.02,17.02,17.22-.04,38.4,0c21.08.04,38.1,17.21,38.06,38.43-.04,21-17.21,38.05-38.32,38.02C17.04,76.42-.02,59.27,0,38.1ZM38.26,5.95c-17.86-.02-32.27,14.33-32.31,32.17-.04,17.94,14.33,32.35,32.25,32.37,17.87.02,32.23-14.28,32.3-32.17.07-17.92-14.31-32.35-32.24-32.37Z"/>
                    <path { ... inheritColor ? {} : {style: {fill: '#1A3061'}}} d="m34.1,43.95c-.12-.12-.18-.26-.18-.43v-1.62c0-1.68.43-3.04,1.29-4.07.86-1.03,2.1-2.11,3.71-3.26,1.31-.91,2.29-1.72,2.93-2.42s.96-1.57.96-2.58c0-1.08-.4-1.95-1.21-2.6-.81-.66-1.87-.99-3.18-.99-1.42,0-2.55.35-3.41,1.06-.86.71-1.29,1.62-1.29,2.73v.81c0,.37-.2.56-.61.56l-5.91-.25c-.17,0-.31-.06-.43-.18-.12-.12-.18-.26-.18-.43v-.5c0-2.02.51-3.81,1.52-5.38,1.01-1.57,2.42-2.78,4.22-3.64,1.8-.86,3.88-1.29,6.24-1.29s4.3.41,6.04,1.24c1.73.83,3.07,1.98,4.02,3.46.94,1.48,1.41,3.2,1.41,5.15,0,1.48-.26,2.75-.78,3.81-.52,1.06-1.15,1.93-1.89,2.6-.74.67-1.68,1.42-2.83,2.22-1.21.84-2.11,1.57-2.68,2.17s-.86,1.35-.86,2.22v1.16c0,.17-.06.31-.18.43-.12.12-.26.18-.43.18h-5.86c-.17,0-.31-.06-.43-.18Zm.3,10.36c-.76-.76-1.14-1.71-1.14-2.85s.37-2.09,1.11-2.83,1.7-1.11,2.88-1.11,2.13.36,2.85,1.09c.72.72,1.09,1.68,1.09,2.85s-.37,2.1-1.11,2.85c-.74.76-1.68,1.14-2.83,1.14s-2.1-.38-2.85-1.14Z"/>
                </svg>
                </span>  
            );
    }
}