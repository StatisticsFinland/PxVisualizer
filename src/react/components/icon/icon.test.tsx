import React from 'react';
import { render } from '@testing-library/react';
import { Icon} from './icon';

it('renders correctly', () => {
    const { asFragment } = render(<Icon icon={'Times'} />);
    expect(asFragment()).toMatchSnapshot();
});
