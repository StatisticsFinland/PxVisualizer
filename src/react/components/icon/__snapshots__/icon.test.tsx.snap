// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<DocumentFragment>
  <svg
    aria-hidden="true"
    height="26.97mm"
    viewBox="0 0 76.44 76.44"
    width="26.97mm"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="m0,38.09C.02,17.02,17.22-.04,38.39,0c21.08.04,38.09,17.21,38.05,38.42-.04,21-17.21,38.04-38.31,38.01C17.04,76.41-.02,59.26,0,38.09ZM38.25,5.95c-17.86-.02-32.27,14.33-32.31,32.17-.04,17.94,14.33,32.35,32.25,32.37,17.87.02,32.23-14.28,32.3-32.17.07-17.92-14.31-32.35-32.24-32.37Z"
      style="fill: #1A3061;"
    />
    <path
      d="m38.4,34c2.9-2.94,5.58-5.67,8.27-8.39.93-.94,2.13-1.38,3.25-.73.8.47,1.7,1.46,1.78,2.29.09.92-.48,2.12-1.14,2.85-2.18,2.38-4.54,4.6-6.83,6.89-.35.35-.68.71-1.07,1.12,2.67,2.68,5.35,5.26,7.88,7.99.67.72,1.28,1.95,1.16,2.84-.12.88-1.05,1.9-1.89,2.37-1.03.57-2.22.16-3.12-.75-2.31-2.33-4.65-4.63-6.96-6.97-.38-.39-.66-.87-1.36-1.8-.71.95-1.11,1.64-1.65,2.19-2.12,2.17-4.27,4.32-6.44,6.44-1.41,1.38-3.16,1.46-4.38.28-1.26-1.22-1.2-3.02.24-4.48,2.27-2.31,4.56-4.58,6.87-6.85.38-.38.88-.64,1.42-1.03-2.89-2.89-5.5-5.47-8.07-8.1-.53-.55-1.05-1.24-1.24-1.96-.35-1.32.14-2.48,1.38-3.16,1.28-.71,2.53-.5,3.56.5,2.32,2.26,4.59,4.55,6.88,6.84.46.46.86.96,1.47,1.64Z"
      style="fill: #1A3061;"
    />
  </svg>
</DocumentFragment>
`;
